src/components/DataTable/DataTable.tsx

```diff
@@ -98,7 +116,18 @@ export const MoonstoneTable = <T extends Record<string, unknown>>({
                     const adaptedRow = isStructured ? adaptRowForTableBodyCell(row) : undefined;
 
                     return (
-                        <TableRow key={row.id}>
+                        <TableRow
+                            key={row.id}
+                            isSelected={enableRowSelection ? row.getIsSelected() : undefined}
```

Comment:
I don't know if we really need the prop `isSelected` at the `TableRow` level. 
The only purpose of this prop is to add an unused CSS class (or at least in one test).